extensions/kubernetes/controllerservice/KubernetesControllerService.cpp

```diff
@@ -0,0 +1,212 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "KubernetesControllerService.h"
+
+#include <vector>
+
+extern "C" {
+#include "config/incluster_config.h"
+#include "config/kube_config.h"
+#include "include/apiClient.h"
+#include "api/CoreV1API.h"
+}
+
+#include "core/Resource.h"
+#include "core/logging/LoggerConfiguration.h"
+#include "Exception.h"
+#include "utils/gsl.h"
+
+namespace org::apache::nifi::minifi::controllers {
+
+class KubernetesControllerService::APIClient {
+ public:
+  explicit APIClient(core::logging::Logger& logger);
```

Comment:
I think it's more readable this way: if I see `ApiClient`, I would assume it has something to do with bees.  I can see the point that it would be friendlier to automated refactoring, but to me, readability now is more important than automated-refactorability maybe in the future.

Reply:
One more point: [`IXMLDOMDocument`](https://docs.microsoft.com/en-us/previous-versions/windows/desktop/ms756987(v=vs.85)). youlosetheseparationofwordswhicharguablyhurtsreadabilitybutifthisisnotconvincingeitherthenwecanleaveitasis

Reply:
That wasn't me! You need to complain to <strike>Bill Gates</strike><strike>Steve Ballmer</strike>Satya Nadella. :window::window::window: